src/components/QuickOpenModal.js

```diff
@@ -134,35 +135,39 @@ export class QuickOpenModal extends Component<Props, State> {
       results = variables;
     }
     if (query === "@" || query === "#") {
-      return this.setState({ results });
+      return this.setState({ results, isLoading: false });
     }
 
-    this.setState({ results: filter(results, query.slice(1)) });
+    this.setState({
+      results: filter(results, query.slice(1)), isLoading: false
+    });
   };
 
   searchShortcuts = (query: string) => {
     const results = formatShortcutResults();
     if (query == "?") {
-      this.setState({ results });
+      this.setState({ results, isLoading: false  });
     } else {
-      this.setState({ results: filter(results, query.slice(1)) });
+      this.setState({
+        results: filter(results, query.slice(1)), isLoading: false
+      });
     }
   };
 
   showTopSources = () => {
     const { tabs, sources } = this.props;
     if (tabs.length > 0) {
       this.setState({
-        results: sources.filter(source => tabs.includes(source.url))
+        results: sources.filter(source => tabs.includes(source.url)), isLoading: false
       });
     } else {
-      this.setState({ results: sources.slice(0, 100) });
+      this.setState({ results: sources.slice(0, 100), isLoading: false });
     }
   };
 
   updateResults = (query: string) => {
     if (this.isGotoQuery()) {
-      return;
+      return this.setState({ isLoading: false });
```

Comment:
hmm, re isLoading... i dont think we need it. Let me explain

the only time we're really loading is when `symbols` are not fetched. We can check this.props.symbols and if it doesn't have any functions it is still loading otherwise we're not

onChange and updateResults dont matter because it's instant